diff -up chromium-59.0.3071.86/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h.gcc7 chromium-59.0.3071.86/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h
--- chromium-59.0.3071.86/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h.gcc7	2017-06-07 16:30:12.351173420 -0400
+++ chromium-59.0.3071.86/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h	2017-06-07 16:32:43.005163948 -0400
@@ -5,6 +5,7 @@
 #include "platform/PlatformExport.h"
 #include "platform/wtf/ThreadSpecific.h"
 
+#include <functional>
 #include <memory>
 
 namespace gpu {
diff -up chromium-59.0.3071.86/v8/src/objects-body-descriptors.h.gcc7 chromium-59.0.3071.86/v8/src/objects-body-descriptors.h
--- chromium-59.0.3071.86/v8/src/objects-body-descriptors.h.gcc7	2017-06-05 15:04:29.000000000 -0400
+++ chromium-59.0.3071.86/v8/src/objects-body-descriptors.h	2017-06-07 16:30:12.352173401 -0400
@@ -99,7 +99,7 @@ class FixedBodyDescriptor final : public
 
   template <typename StaticVisitor>
   static inline void IterateBody(HeapObject* obj, int object_size) {
-    IterateBody(obj);
+    IterateBody<StaticVisitor>(obj);
   }
 };
 
diff -up chromium-59.0.3071.86/v8/src/objects/hash-table.h.gcc7 chromium-59.0.3071.86/v8/src/objects/hash-table.h
--- chromium-59.0.3071.86/v8/src/objects/hash-table.h.gcc7	2017-06-07 16:35:26.052900374 -0400
+++ chromium-59.0.3071.86/v8/src/objects/hash-table.h	2017-06-07 16:35:53.982340480 -0400
@@ -135,22 +135,8 @@ class HashTable : public HashTableBase {
  public:
   typedef Shape ShapeT;
 
-  // Wrapper methods
-  inline uint32_t Hash(Key key) {
-    if (Shape::UsesSeed) {
-      return Shape::SeededHash(key, GetHeap()->HashSeed());
-    } else {
-      return Shape::Hash(key);
-    }
-  }
-
-  inline uint32_t HashForObject(Key key, Object* object) {
-    if (Shape::UsesSeed) {
-      return Shape::SeededHashForObject(key, GetHeap()->HashSeed(), object);
-    } else {
-      return Shape::HashForObject(key, object);
-    }
-  }
+  inline uint32_t Hash(Key key);
+  inline uint32_t HashForObject(Key key, Object* object);
 
   // Returns a new HashTable object.
   MUST_USE_RESULT static Handle<Derived> New(
diff -up chromium-59.0.3071.86/v8/src/objects.h.gcc7 chromium-59.0.3071.86/v8/src/objects.h
diff -up chromium-59.0.3071.86/v8/src/objects-inl.h.gcc7 chromium-59.0.3071.86/v8/src/objects-inl.h
--- chromium-59.0.3071.86/v8/src/objects-inl.h.gcc7	2017-06-05 15:04:29.000000000 -0400
+++ chromium-59.0.3071.86/v8/src/objects-inl.h	2017-06-07 16:30:12.477171021 -0400
@@ -46,6 +46,25 @@
 namespace v8 {
 namespace internal {
 
+template <typename Derived, typename Shape, typename Key>
+uint32_t HashTable<Derived, Shape, Key>::Hash(Key key) {
+  if (Shape::UsesSeed) {
+    return Shape::SeededHash(key, GetHeap()->HashSeed());
+  } else {
+    return Shape::Hash(key);
+  }
+}
+
+template <typename Derived, typename Shape, typename Key>
+uint32_t HashTable<Derived, Shape, Key>::HashForObject(Key key,
+                                                       Object* object) {
+  if (Shape::UsesSeed) {
+    return Shape::SeededHashForObject(key, GetHeap()->HashSeed(), object);
+  } else {
+    return Shape::HashForObject(key, object);
+  }
+}
+
 PropertyDetails::PropertyDetails(Smi* smi) {
   value_ = smi->value();
 }
