diff -up chromium-59.0.3071.86/cc/blink/web_layer_impl.h.setopaque chromium-59.0.3071.86/cc/blink/web_layer_impl.h
--- chromium-59.0.3071.86/cc/blink/web_layer_impl.h.setopaque	2017-06-07 15:48:52.240620197 -0400
+++ chromium-59.0.3071.86/cc/blink/web_layer_impl.h	2017-06-07 16:25:48.311448200 -0400
@@ -69,7 +69,7 @@ class CC_BLINK_EXPORT WebLayerImpl : pub
   blink::WebBlendMode BlendMode() const override;
   void SetIsRootForIsolatedGroup(bool root) override;
   bool IsRootForIsolatedGroup() override;
-  void SetOpaque(bool opaque) override;
+  CC_BLINK_EXPORT void SetOpaque(bool opaque) override;
   bool Opaque() const override;
   void SetPosition(const blink::WebFloatPoint& position) override;
   blink::WebFloatPoint GetPosition() const override;
