diff -up chromium-59.0.3071.86/build/toolchain/linux/BUILD.gn.nullfix chromium-59.0.3071.86/build/toolchain/linux/BUILD.gn
--- chromium-59.0.3071.86/build/toolchain/linux/BUILD.gn.nullfix	2017-06-05 15:03:01.000000000 -0400
+++ chromium-59.0.3071.86/build/toolchain/linux/BUILD.gn	2017-06-07 15:32:43.531911624 -0400
@@ -31,6 +31,7 @@ gcc_toolchain("arm64") {
   ld = cxx
   readelf = "${toolprefix}readelf"
   nm = "${toolprefix}nm"
+  extra_cppflags = "-fno-delete-null-pointer-checks"
 
   toolchain_args = {
     current_cpu = "arm64"
@@ -49,6 +50,7 @@ gcc_toolchain("arm") {
   ld = cxx
   readelf = "${toolprefix}readelf"
   nm = "${toolprefix}nm"
+  extra_cppflags = "-fno-delete-null-pointer-checks"
 
   toolchain_args = {
     current_cpu = "arm"
@@ -91,6 +93,7 @@ gcc_toolchain("x86") {
   nm = "nm"
   ar = "ar"
   ld = cxx
+  extra_cppflags = "-fno-delete-null-pointer-checks"
 
   # Output linker map files for binary size analysis.
   enable_linker_map = true
@@ -136,6 +139,7 @@ gcc_toolchain("x64") {
   nm = "nm"
   ar = "ar"
   ld = cxx
+  extra_cppflags = "-fno-delete-null-pointer-checks"
 
   # Output linker map files for binary size analysis.
   enable_linker_map = true
@@ -161,6 +165,7 @@ gcc_toolchain("mipsel") {
   ld = cxx
   readelf = "mipsel-linux-gnu-readelf"
   nm = "mipsel-linux-gnu-nm"
+  extra_cppflags = "-fno-delete-null-pointer-checks"
 
   toolchain_args = {
     cc_wrapper = ""
