diff --color -Naru a/setup.py b/setup.py
--- a/setup.py	2022-07-07 10:31:36.059505358 +0200
+++ b/setup.py	2022-07-07 10:32:02.049505280 +0200
@@ -13,7 +13,7 @@
     author='eliep',
     author_email='eprudhomme@gmail.com',
     license='MIT',
-    packages=setuptools.find_packages(),
+    packages=setuptools.find_packages(exclude=["test", "test.*"]),
     classifiers=[
         "Programming Language :: Python :: 3",
         "License :: OSI Approved :: MIT License",
