diff -Naru a/package/contents/ui/config/configAppearance.qml b/package/contents/ui/config/configAppearance.qml
--- a/package/contents/ui/config/configAppearance.qml	2019-12-20 14:06:06.273980311 +0100
+++ b/package/contents/ui/config/configAppearance.qml	2019-12-20 14:06:48.943980929 +0100
@@ -17,6 +17,7 @@
 		
 		PlasmaComponents.Label {
 			text: i18n("Font size:")
+			color: myPalette.textColor
 		}
 		
 		SpinBox {
@@ -29,6 +30,7 @@
 		
 		PlasmaComponents.Label {
 			text: i18n("Online icon:")
+			color: myPalette.textColor
 		}
 		
 		IconPicker {
@@ -41,6 +43,7 @@
 		
 		PlasmaComponents.Label {
 			text: i18n("Offline icon:")
+			color: myPalette.textColor
 		}
 		
 		IconPicker {
diff -Naru a/package/contents/ui/config/configGeneral.qml b/package/contents/ui/config/configGeneral.qml
--- a/package/contents/ui/config/configGeneral.qml	2019-12-20 14:06:06.273980311 +0100
+++ b/package/contents/ui/config/configGeneral.qml	2019-12-20 14:06:48.943980929 +0100
@@ -173,6 +173,7 @@
 				
 				PlasmaComponents.Label {
 					text: "Name:"
+					color: myPalette.textColor 
 				}
 				
 				TextField {
@@ -183,6 +184,7 @@
 				
 				PlasmaComponents.Label {
 					text: "Host name:"
+					color: myPalette.textColor
 				}
 				
 				TextField {
@@ -193,6 +195,7 @@
 				
 				PlasmaComponents.Label {
 					text: i18n("Refresh rate:")
+					color: myPalette.textColor
 				}
 				
 				SpinBox {
@@ -205,6 +208,7 @@
 				
 				PlasmaComponents.Label {
 					text: i18n("Check method:")
+					color: myPalette.textColor
 				}
 				
 				ComboBox {
diff -Naru a/package/contents/ui/main.qml b/package/contents/ui/main.qml
--- a/package/contents/ui/main.qml	2019-12-20 14:06:06.273980311 +0100
+++ b/package/contents/ui/main.qml	2019-12-20 14:06:48.943980929 +0100
@@ -22,6 +22,11 @@
 	
 // 	Plasmoid.preferredRepresentation: Plasmoid.compactRepresentation
 	Plasmoid.toolTipTextFormat: Text.RichText
+
+	SystemPalette { 
+		id: myPalette; 
+		colorGroup: SystemPalette.Active 
+	}
 	
 	ServersModel {
 		id: serversModel
