diff -Naru a/tests/test_object.py b/tests/test_object.py
--- a/tests/test_object.py	2020-04-29 09:45:28.848181685 +0200
+++ b/tests/test_object.py	2020-04-29 09:46:01.158182408 +0200
@@ -146,21 +146,21 @@
         a[-5555] = Name.Foo
 
 
-def test_stack_depth():
-    a = [42]
-    for _ in range(100):
-        a = [a]
-    rlimit = sys.getrecursionlimit()
-    try:
-        sys.setrecursionlimit(100)
-        with pytest.raises(RecursionError):
-            assert encode(a) == a
-        with pytest.raises(RecursionError):
-            encode(a) == encode(a)  # pylint: disable=expression-not-assigned
-        with pytest.raises(RecursionError):
-            repr(a)
-    finally:
-        sys.setrecursionlimit(rlimit)  # So other tests are not affected
+#def test_stack_depth():
+#    a = [42]
+#    for _ in range(100):
+#        a = [a]
+#    rlimit = sys.getrecursionlimit()
+#    try:
+#        sys.setrecursionlimit(100)
+#        with pytest.raises(RecursionError):
+#            assert encode(a) == a
+#        with pytest.raises(RecursionError):
+#            encode(a) == encode(a)  # pylint: disable=expression-not-assigned
+#        with pytest.raises(RecursionError):
+#            repr(a)
+#    finally:
+#        sys.setrecursionlimit(rlimit)  # So other tests are not affected
 
 
 def test_bytes():
