diff --git a/CMakeLists.txt b/CMakeLists.txt
index 994c3b4..1940bcd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,8 +18,6 @@ else()
     list(APPEND CMAKE_MODULE_PATH ${VCM_MODULE_PATH})
 endif()
 include(GitVersion)
-determine_fallback_version(${CMAKE_SOURCE_DIR}/cmake)
-include(${CMAKE_SOURCE_DIR}/cmake/FallbackVersion.cmake)
 include(Format)
 
 # Add 'make dist' command for creating release tarball
